python/cudf/cudf/core/column/decimal.py

```diff
@@ -366,6 +366,20 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     else:
         raise NotImplementedError()
 
+    # potentially reduce scale and precision
+    if precision > cudf.Decimal128Dtype.MAX_PRECISION:
+        if op in {"__add__", "__sub__"}:
```

Comment:
Can we combine the handling for each specific op to the section of code above so we do all the `scale` and `precision` calculations for each op happen in once place?